src/main/java/net/minecraftforge/client/gui/config/ConfigScreen.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2020.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client.gui.config;
+
+import com.mojang.blaze3d.matrix.MatrixStack;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.DialogTexts;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.widget.Widget;
+import net.minecraft.client.gui.widget.button.Button;
+import net.minecraft.util.text.IFormattableTextComponent;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraftforge.client.gui.config.ConfigElementList.ConfigElement;
+import net.minecraftforge.fml.client.gui.GuiUtils;
+import net.minecraftforge.fml.client.gui.widget.ExtendedButton;
+import net.minecraftforge.fml.client.gui.widget.UnicodeGlyphButton;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+
+/**
+ * This class standardises the layout/look/feel of a config screen.
+ * It passes necessary events through to it's {@link ConfigElementList}.
+ * <p>
+ * TODO:
+ * - Get someone's review on changes to ForgeConfigSpec for comments
+ * - Make ConfigScreen header size be smaller if it has a null subTitle?
+ * - Translucent scissored list?
+ * - Do I need to deal with sub-configs?
+ * - Fix lists
+ * -> Change indices instead of reloading the entire view
+ * -> Drag and drop movement?
+ * -> Maybe try and refactor it all to be less confusing and twisted
+ * - Client -> Server syncing?
```

Comment:
If a server config is being changed, it needs to be synced to the server and all clients. It should also (IMO) only be changeable by owners. So yes, we need syncing, but we already have a packet for this (although we might need a new one for in-game changes to config). Most likely it should also resync when the file is changed by the filesystem, which IIRC it doesn't do right now as well.

Reply:
> If a server config is being changed, it needs to be synced to the server and all clients.
> Most likely it should also resync when the file is changed by the filesystem, which IIRC it doesn't do right now as well.

Yes, I was assuming that I would just need to send a packet to the server, it would save & reload the config and then that the server -> all clients syncing was already implemented. If it isn't, it shouldn't be too hard to implement.

> It should also (IMO) only be changeable by owners

Definitely, I was thinking changeable by OPs

> we already have a packet for this

We have one for Server -> Client sync during the handshake, I think this one could be repurposed for in-game updates but I'm not sure... We definitely need a new one for Client -> Server config update request.
I've left out the code for syncing right now because I know it will be a bit of work to implement.

Reply:
> Definitely, I was thinking changeable by OPs
Would be cool if you would utilize the forge permission api for that, so that people are encouraged to use it more